core/java/iesi-core/src/test/java/io/metadew/iesi/script/action/http/HttpExecuteRequestTest.java

```diff
@@ -0,0 +1,453 @@
+
+        ActionParameter requestActionParameter = createActionParameter("request", "request");
+        ActionParameter queryParametersActionParameter = createActionParameter("queryParameters", queryParameters);
+        Action action = createAction(requestActionParameter, queryParametersActionParameter);
+
+        when(actionExecution.getAction())
+                .thenReturn(action);
+
+        mockResolvedValues(requestActionParameter, queryParametersActionParameter);
+
+        HttpComponent httpComponent = createBaseComponent(new ArrayList<>(), new ArrayList<>());
+        mockGetAndTraceHttpComponent(httpComponent);
+
+        HttpExecuteRequest httpExecuteRequest = new HttpExecuteRequest(executionControl, scriptExecution, actionExecution);
+        httpExecuteRequest.prepare();
+
+        HttpRequestBase httpRequest = httpExecuteRequest.getHttpRequest().getHttpRequest();
+
+        if (queryParameters.equals("status=sold")) {
+            assertThat(httpRequest.getURI()).hasToString("https://hostendpoint?status=sold");
+        } else if (queryParameters.equals("status=sold,name=name")) {
+            assertThat(httpRequest.getURI()).hasToString("https://hostendpoint?name=name&status=sold");
+        }
+
+    }
+
+
+    @Test
+    void prepareNoHeadersAndWrongQueries() {
+
+        ActionParameter requestActionParameter = createActionParameter("request", "request");
+        ActionParameter queryParametersActionParameter = createActionParameter("queryParameters", "status=====sold,namename");
+        Action action = createAction(requestActionParameter, queryParametersActionParameter);
+
+        when(actionExecution.getAction())
+                .thenReturn(action);
+
+        mockResolvedValues(requestActionParameter, queryParametersActionParameter);
+
+        HttpComponent httpComponent = createBaseComponent(new ArrayList<>(), new ArrayList<>());
+        mockGetAndTraceHttpComponent(httpComponent);
+
+        HttpExecuteRequest httpExecuteRequest = new HttpExecuteRequest(executionControl, scriptExecution, actionExecution);
+
+        assertThrows(KeyValuePairException.class, httpExecuteRequest::prepare);
+    }
+
+    @Test
+    void prepareNoStatusCode() throws HttpRequestBuilderException, URISyntaxException {
+
+        ActionParameter requestActionParameter = createActionParameter("request", "request");
+        Action action = createAction(requestActionParameter);
+
+        when(actionExecution.getAction())
+                .thenReturn(action);
+
+        mockResolvedValues(requestActionParameter);
+
+        HttpComponent httpComponent = createBaseComponent(new ArrayList<>(), new ArrayList<>());
+        mockGetAndTraceHttpComponent(httpComponent);
+
+        HttpExecuteRequest httpExecuteRequest = new HttpExecuteRequest(executionControl, scriptExecution, actionExecution);
+        httpExecuteRequest.prepare();
+
+
+        assertThat(httpExecuteRequest.getExpectedStatusCodes()).isEmpty();
```

Comment:
test this in the first unit test, the 'use case' is the same: only request defined